R/rdeephaven/R/client_wrapper.R

```diff
@@ -1,110 +1,249 @@
-    #' @return TableHandle reference to the new table.
-    import_table = function(table_object) {
-      table_object_class <- class(table_object)
-      if (table_object_class[[1]] == "data.frame") {
-        return(TableHandle$new(private$df_to_dh_table(table_object)))
+#' @export
+setClass(
+  "Client",
+  representation(
+    .internal_rcpp_object = "Rcpp_INTERNAL_Client"
+  )
+)
+
+setGeneric(
+  "connect",
+  function(target, ...) {
+    return(standardGeneric("connect"))
+  },
+  signature = c("target")
+)
+
+#' @export
+setMethod(
+  "connect",
+  signature = c(target = "character"),
+  function(target,
+           auth_type = "anonymous",
+           auth_token_pair = "",
+           session_type = "python",
+           use_tls = FALSE,
+           tls_root_certs = "",
+           int_option = "",
+           string_option = "",
+           extra_header = "") {
+    options <- new(INTERNAL_ClientOptions)
+
+    # check if auth_type needs to be changed and set credentials accordingly
+    if (auth_type != "anonymous") {
+      if (auth_type == "basic") {
+        if (auth_token_pair != "") {
+          verify_string("auth_token_pair", auth_token_pair, TRUE)
+          username_password <- strsplit(auth_token_pair, ":", fixed = TRUE)
+          options$set_basic_authentication(username_password[1], username_password[2])
+        } else {
+          stop("Basic authentication was requested, but no 'auth_token_pair' was provided.")
+        }
+      } else if (auth_type == "custom") {
+        if (auth_token_pair != "") {
+          verify_string("auth_token_pair", auth_token_pair, TRUE)
+          key_value <- strsplit(auth_token_pair, ":", fixed = TRUE)
+          options$set_custom_authentication(key_value[1], key_value[2])
+        } else {
+          stop("Custom authentication was requested, but no 'auth_token_pair' was provided.")
+        }
+      } else {
+        stop(paste0("'auth_type' must be 'anonymous', 'basic', or 'custom', but got ", auth_type, " instead."))
       }
-      if (table_object_class[[1]] == "tbl_df") {
-        return(TableHandle$new(private$tibble_to_dh_table(table_object)))
-      } else if (table_object_class[[1]] == "RecordBatchReader") {
-        return(TableHandle$new(private$rbr_to_dh_table(table_object)))
-      } else if ((length(table_object_class) == 4 &&
-        table_object_class[[1]] == "Table" &&
-        table_object_class[[3]] == "ArrowObject")) {
-        return(TableHandle$new(private$arrow_to_dh_table(table_object)))
+    }
+
+    # set session type if a valid session type is provided
+    if ((session_type == "python") || (session_type == "groovy")) {
+      options$set_session_type(session_type)
+    } else {
+      stop(paste0("'session_type' must be 'python' or 'groovy', but got ", session_type, " instead."))
+    }
+
+    # if tls is requested, set it and set the root_certs if provided
+    if (use_tls != FALSE) {
+      if (use_tls == TRUE) {
```

Comment:
???.  

If you are concerned about non-boolean stuff, you should have an assertion at the start of the function to avoid this.